browser/skus/skus_service_factory.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/skus/skus_service_factory.h"
+
+#include "base/feature_list.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/skus/browser/skus_service_impl.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "brave/components/skus/common/features.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+
+namespace skus {
+
+// static
+SkusServiceFactory* SkusServiceFactory::GetInstance() {
+  return base::Singleton<SkusServiceFactory>::get();
+}
+
+// static
+mojo::PendingRemote<mojom::SkusService> SkusServiceFactory::GetForContext(
+    content::BrowserContext* context) {
+  if (!context || !base::FeatureList::IsEnabled(skus::features::kSkusFeature)) {
```

Comment:
This is not correct. `context` will never be null so that's not an issue, but you need to check that `GetInstance()->GetServiceForBrowserContext(context, true)` is not null and that's all you need to check. The feature check is already in `BuildServiceInstanceFor`

Reply:
see https://github.com/brave/brave-core/pull/10847/files#diff-092da8e8150cc48e0f400678983296898a026dcee82479a5eb09f3f07310b8b7R23 which is correct

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/d634f38c925efbd98b7fe4cb274ae77d102b431d